migrate-from-non-mysql.md

```diff
@@ -0,0 +1,186 @@
+
+
+--
+-- PostgreSQL database dump complete
+--
+
+```
+
+So we need to filter out the `CREATE TABLE` statement and modify it to:
+
+- Use `AUTO_INCREMENT` or `AUTO_RANDOM` instead of the sequence
+- Set the `id` column to be the `PRIMARY KEY`
+- Replace the `character varying` with `VARCHAR`
+- Change the name from `public.t1` to `t1`.
+
+```
+CREATE TABLE t1 (
+    id integer NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(255)
+)
+```
+
+Now we dump the data to a tab separated file
+
+```
+test=# COPY t1 TO '/tmp/t1.dump' WITH (FORMAT CSV, DELIMITER E'\t');
+COPY 3
+test=# \! cat /tmp/t1.dump
+1	foo
+2	bar
+3	baz
+```
+
+Now we create the schema, load the data and finally verify the content:
+
+```
+tidb> CREATE TABLE t1 (
+   ->     id integer NOT NULL AUTO_INCREMENT PRIMARY KEY,
+   ->     name VARCHAR(255)
+   -> );
+Query OK, 0 rows affected (0.2136 sec)
+tidb> LOAD DATA LOCAL INFILE '/tmp/t1.dump' INTO TABLE t1;
+Query OK, 3 rows affected (0.0509 sec)
+
+Records: 3  Deleted: 0  Skipped: 0  Warnings: 0
+tidb> TABLE t1;
++----+------+
+| id | name |
++----+------+
+|  1 | foo  |
+|  2 | bar  |
+|  3 | baz  |
++----+------+
+3 rows in set (0.0037 sec)
+```
+
+## Continues migration
+
+Continues migration means that changes that happen over time on the source database are applied on the target database. This means that the source and target databases stay in sync, but there can be a delay with applying the changes on the target database.
+
+There are multiple situations where this is useful:
+
+- When migrating a large database you dump all data and then import the data on the target. This may take many hours or even days. If you now setup continues migration you can bring back this time to less than an hour. This allows you to compare the results and performance of your read-only queries between the source and target databases. And when you are ready to switch to the target database it takes less time to sync reducing the downtime your application may have during migration.
+
+- When combining data from multiple database systems into one. By combining data from multiple systems or multiple instances of an application it allows you to run analytical queries against them. You may also be able to delete this data from the source system once it is archived into the target system if you setup the migration to filter out deletes or otherwise prevent these from reaching your target database.
+
+There are multiple tools that can be used for continues migration:
+
+- Apache Flink
+- Apache Kafka wit Kafka Connect
+
+Both Flink and Kafka Connect rely on [Debezium](https://debezium.io) for most of the database streaming implementations.
+
+> **Note:**
+> When using continues migration DML (`INSERT`, `UPDATE`, `DELETE`, etc) is usually fine but DDL (`CREATE TABLE`,`ALTER TABLE`, etc) can be problematic. Depending on the tools and configuration the tool could stop to apply changes on the target database and require an operator to step in.
```

Comment:
DDL accounts for statements like adding/removing indexes. Did you want to target a subset and say "schema changes"?
```suggestion
> When using continues migration DML (`INSERT`, `UPDATE`, `DELETE`, etc) is usually fine but schema changes are likely to be problematic. Depending on the tools and configuration the tool could stop to apply changes on the target database and require an operator to step in.
```